examples/ecdsa/methods/guest/src/bin/benchmark.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{hint::black_box, ops::Add};
+
+use hex_literal::hex;
+use k256::{
+    elliptic_curve::{
+        bigint::{Encoding, U256},
+        ops::{Invert, LinearCombination, Reduce},
+    },
+    FieldElement, ProjectivePoint, Scalar,
+};
+use risc0_zkvm::guest::env;
+
+fn bench(name: &str, func: impl Fn()) {
+    // Run the inner function twice, only logging the cycles in the second run, in
+    // order to ensure the code and memory has been paged-in.
+    bench_inner(name, &func, true);
+    bench_inner(name, &func, false);
+}
+
+#[inline(never)]
+fn bench_inner(name: &str, func: impl FnOnce(), silent: bool) {
+    black_box({
+        let start = env::get_cycle_count();
+
+        black_box(func());
+
+        let end = env::get_cycle_count();
```

Comment:
Unforunately using cycle count in this way won't be very accurate, due to the difference between the way pages are emulated in the executor and performed in the prover. I'm not sure how much we can trust these numbers in this way, let's chat about this offline at some point so I can provide deeper context.